docs/DeveloperGuide.md

```diff
@@ -1,29 +1,265 @@
 # Developer Guide
 
+## Table of Contents
+
+* [Design](#design)
+  * [Architecture](#architecture)
+    * [Main components](#main-components)
+    * [Component interaction](#component-interaction)
+  * [UI components](#ui-components)
+  * [EconoCraftLogic components](#econocraftlogic-components)
+  * [MiniGame components](#miniGame-components)
+* [Implementation](#implementation)
+  * [MiniGame - Typing Game](#miniGame---typing-Game)
+  * [MiniGame - Tic Tac Toe](#miniGame---tic-tac-toe)
+  * [MiniGame - True or False](#miniGame---true-or-false)
+* [Product scope](#product-scope)
+  * [Target user playerProfile](#target-user-playerprofile)
+  * [Value proposition](#value-proposition)
+* [User Stories](#user-stories)
+* [Non-Functional Requirements](#non-functional-requirements)
+* [Glossary](#glossary)
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
+## Design
+
+### Overview of GameFlow
+![GameFlow.jpg](UML%20diagram%2FGameFlow.jpg)
```

Comment:
I like the flow chart. I think it may be more readable if the size of the picture was reduced?